.github/workflows/test.yml

```diff
@@ -1,32 +1,77 @@
 name: Test
-on: [push, pull_request]
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+    branches:
+      - master
+      - main
+
 jobs:
   test:
+    permissions:
+      contents: read      # for actions/checkout to fetch code
+    timeout-minutes: 10
+
     strategy:
       matrix:
-        go-versions: [1.12.x, 1.13.x, 1.14.x]
-        platform: [ubuntu-latest]
-    runs-on: ${{ matrix.platform }}
+        # test against the "oldest" supported version and the current version
+        # of go. Go 1.17 is kept in this matrix as it is the minimum version
+        # specified in go.mod, and maintaining compatibility with go 1.17 is
+        # currently not much of a burden. Most projects using this module are
+        # using newer versions than that, so we can drop the old version if
+        # it becomes too much of a burden.
+        go-version: [1.17.x, 1.19.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
     steps:
     - name: Install Go
-      uses: actions/setup-go@v1
+      uses: actions/setup-go@v3
       with:
         go-version: ${{ matrix.go-version }}
     - name: Checkout code
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
+    - name: go mod tidy
+      run: |
+        go mod tidy
+        git diff --exit-code
     - name: Ensure IPVS module
-      run: sudo modprobe ip_vs
+      run: |
```

Comment:
I don't think there's a need for the multiline string syntax when we only have one line

Reply:
Ah, yeah, I made the change as I was syncing with other repositories (made it easier to compare).

Happy to revert if you prefer though (let me know)

Reply:
That justification makes sense; ultimately using composite actions among the repos that are very similar (e.g. these small modules) is probably something we should explore.